dm/master/election.go

```diff
@@ -55,21 +57,46 @@ func (s *Server) electionNotify(ctx context.Context) {
 				continue
 			}
 
+			failToStart := false
+
 			if leaderInfo.ID == s.cfg.Name {
 				// this member become leader
 				log.L().Info("current member become the leader", zap.String("current member", s.cfg.Name))
 				err := s.scheduler.Start(ctx, s.etcdClient)
 				if err != nil {
 					log.L().Error("scheduler do not started", zap.Error(err))
+					failToStart = true
 				}
 
 				err = s.pessimist.Start(ctx, s.etcdClient)
-				if err != nil {
+				if err != nil && !failToStart {
```

Comment:
if `failToStart` is `true`, should we still try to `start` `pessimist` and `optimist`?

Reply:
fix it in 5911e58